src/pybind/mgr/cephadm/module.py

```diff
@@ -1284,6 +1372,8 @@ def _refresh_host_daemons(self, host) -> Optional[str]:
         return None
 
     def _refresh_host_devices(self, host) -> Optional[str]:
```

Comment:
either rename the function:

```suggestion
    def _refresh_host_devices_and_set_registry_credentials(self, host) -> Optional[str]:
```

or put the code into a better location, please.

Reply:
My plan is to change this section to match your suggestion here (which would remove the need to do a registry login inside the refresh host devices function): https://github.com/ceph/ceph/pull/36012#discussion_r454209571
Haven't gotten to it quite yet but wanted review on the JSON related things in the mean time.